src/client/Microsoft.Identity.Client/MsalServiceExceptionFactory.cs

```diff
@@ -45,12 +46,19 @@ internal class MsalServiceExceptionFactory
                     innerException);
             }
 
+            if (httpResponse.HeadersAsDictionary != null && httpResponse.HeadersAsDictionary.ContainsKey(ThrottleCommon.ThrottleRetryAfterHeaderResponseValue))

```

Comment:
I don't think you care about the existence of a header. You should look for not having a status code is 429 and no json response.

nit: HeadersAsDictionary  is bad property. It actually does a lot of work on the getter.

Reply:
ok, I was looking for the header because the 429 HttpResponse status code enum was not available in .net 4.5. and 4.6.

Reply:
HttpResponse is an enum, you can cast to int and manually check for 429. You can rely on CreateTooManyRequestsNonJsonResponse() for an example HTTP response.